### PR TITLE
cover nn.Conv1d in mkldnn model conversion logic

### PR DESCRIPTION
**Pull request by mingfeima**
current `to_mkldnn` model conversion logic under `torch.utils.mkldnn` does not cover `nn.Conv1d`. This patch fills the gap, using similar logic to `nn.Conv2d`. The model conversion will remove unnecessary memory format reorders of input/output tensors and thus speedup the model.

### CODE REVIEW

File: torch/utils/mkldnn.py [hunk 1]
```diff
@@ -33,6 +33,49 @@ def forward(self, x):
         return y
 
 
+class MkldnnConv1d(torch.jit.ScriptModule):
```
Comment (albanD) [inline torch/utils/mkldnn.py:36]: There seem to be a lot of copy paste with Conv2d.
Would it be possible to have a common subclass here to reduce code duplication?

Comment (mingfeima) [reply torch/utils/mkldnn.py]: Yes, indeed. The logic here is mostly copy paste. I have write a base class `MkldnnConvNd`.

**Comment by mingfeima** [comment]
@albanD please review!

File: torch/utils/mkldnn.py [hunk 2]
```diff
@@ -63,26 +58,51 @@ def __getstate__(self):
...
-        return torch.conv2d(
-            x,
-            self.weight,
-            self.bias,
+
+class MkldnnConv2d(_MkldnnConvNd):
+    def __init__(self, dense_module):
+        super(MkldnnConv2d, self).__init__(dense_module)
+
+        self.register_buffer('weight', torch._C._nn.mkldnn_reorder_conv2d_weight(
```
Comment (albanD) [inline torch/utils/mkldnn.py:89]: Out of curiosity why are these buffers and not parameters? They are not expected to be returned when you do `model.parameter()` ?

**Review by albanD — APPROVED** [review]
Looks good.

**Pull request closed without merging**